### PR TITLE
Migrate and update example from examples repo

### PR DESCRIPTION
**Pull request by JamieDanielson**
## Which problem is this PR solving?

- migrates node-tracing example from examples repo

## Short description of the changes

- simplifies example (no need for third party APIs)
- updated to use local beeline code as dependency, which is installed and linked with `npm run setup`
- removed deprecated methods in favor of new, e.g. `beeline.honeycomb.marshalTraceContext`
- Add custom span and fixed TraceContext example
- set everything to run as `npm start`

### CODE REVIEW

**Review by vreynolds — APPROVED** [review]

**Comment by JamieDanielson** [comment]
@vreynolds requesting a re-review as I tore out some of the original example and added some new stuff in.

File: examples/node-tracing/app.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+/* eslint-env node */
+"use strict";
+const beeline = require("honeycomb-beeline");
+
+const HONEYCOMB_DATASET = `${process.env.HONEYCOMB_DATASET}` || "tracing-example";
```
Comment (vreynolds) [inline examples/node-tracing/app.js:5]: string interpolation results in a literal string "undefined" when there's no env var

```suggestion
const HONEYCOMB_DATASET = process.env.HONEYCOMB_DATASET || "tracing-example";
```

File: examples/node-tracing/app.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+const SERVICE_NAME = `${process.env.SERVICE_NAME}` || "wall";
```
Comment (vreynolds) [inline examples/node-tracing/app.js:6]:
```suggestion
const SERVICE_NAME = process.env.SERVICE_NAME || "wall";
```

File: examples/node-tracing/app.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+const HONEYCOMB_API_KEY = `${process.env.HONEYCOMB_API_KEY}` || "abc123";
```
Comment (vreynolds) [inline examples/node-tracing/app.js:7]:
```suggestion
const HONEYCOMB_API_KEY = process.env.HONEYCOMB_API_KEY || "abc123";
```

**Review by vreynolds — CHANGES_REQUESTED** [review]

**Review by vreynolds — APPROVED** [review]
✨🍾✨

**Pull request merged**